src/Microsoft.Health.Fhir.TaskManagement/ITaskConsumer.cs

```diff
@@ -0,0 +1,22 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Fhir.TaskManagement
+{
+    public interface ITaskConsumer
+    {
+        Task<TaskInfo> CompleteAsync(string taskId, TaskResultData result, string runId, CancellationToken cancellationToken);
+
+        Task<IReadOnlyCollection<TaskInfo>> GetNextMessagesAsync(short count, int taskHeartbeatTimeoutThresholdInSeconds, CancellationToken cancellationToken);
```

Comment:
Can `taskHeartbeatTimeoutThresholdInSeconds` just be a timespan?

Reply:
For this threshold we target to have reasonable level precision and looks like seconds is enough (keepalive is at seconds level), not sure is there any special case we might need general timespan?